wireload/screenly-ose
# migrate to ansible & systemd

1. Install.sh - bootstrap script for 
  - install ansible
  - check out latest screenly sources
  - run ansible
2. Install.sh should NEVER EVER be modified
3. Use install.sh for ALL - for installation and for upgrade. This is possible with Ansible.

Install.sh run's ansible playbooks. Now it make like this:

1. system.yml - keep system environment: systemd and other software like xserver and matchbox 
  - install systemd
  - make systemd as init (required for debian wheezy)
  - other preparations migrated from install.sh

2. screenly.yml - keep screenly-related system state
  - copy default config's
  - run migrations
  - enable systemd unit

Systemd issues:
On wheezy (on first installation / migration) screenly.yml playbook  fails on 'systemctl enable xxx.service' because we are doing this on machine booted on sysv init. BUT - it works! Systemctl enable should just make symlink and it make's and fails after that on machine, booted on sysv. It seems that we must write some good message for this.

Systemd goodies:

1. It's really fast!
2. There is no supervisor and wierd initialization
3. Easy log management e.g. journalctl -u screenly-web.service


---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/wireload/screenly-ose/368)
